### PR TITLE
Ahayel/22 unittests

### PR DESCRIPTION
**Pull request by artemgael96**

### CODE REVIEW

**Review by vlad12345123 — APPROVED** [review]

File: exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            _mockConfiguration.Setup(u => u["JWT:ValidIssuer"]).Returns(AuthenticateControllerrFixture.GetJWTValidIssuer());
+            _mockConfiguration.Setup(u => u["JWT:ValidAudience"]).Returns(AuthenticateControllerrFixture.GetJWTValidAudience());
+        }
+
+        [Fact]
+        public async Task FindByNameAsync_LoginModelIsEmpty_AllFieldsAreCorrect()
+        {
+            // Arrange
+            var _controller = new AuthenticateController(_mockUserManager.Object, _mockConfiguration.Object);
+            var loginModel = new LoginModel();
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs]: add email, password

File: exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            _mockUserManager.Setup(u => u.CheckPasswordAsync(It.IsAny<User>(), It.IsAny<string>())).Returns(AuthenticateControllerrFixture.IsCorrectPasswordAsync());
+
+            _mockConfiguration = new Mock<IConfiguration>();
+            _mockConfiguration.Setup(u => u["JWT:Secret"]).Returns(AuthenticateControllerrFixture.GetJWTSecret());
+            _mockConfiguration.Setup(u => u["JWT:ValidIssuer"]).Returns(AuthenticateControllerrFixture.GetJWTValidIssuer());
+            _mockConfiguration.Setup(u => u["JWT:ValidAudience"]).Returns(AuthenticateControllerrFixture.GetJWTValidAudience());
+        }
+
+        [Fact]
+        public async Task FindByNameAsync_LoginModelIsEmpty_AllFieldsAreCorrect()
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs]: rename

File: exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using ExaLearn.Dal.Entities;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Tests.Fixture
+{
+    public static class AuthenticateControllerrFixture
+    {
+        public static async Task<User> GetUserNameAsync() => new User() { UserName = "adminexa@mailnesia.com" };
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs]: password _Test1234

File: exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+        public static async Task<bool> IsCorrectPasswordAsync() => true;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs]: (bool IsCorrect) => IsCorrect

File: exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+            // Act
+            dynamic result = await _controller.Login(loginModel);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(result.StatusCode, 200);
+            var token = result.Value.GetType().GetProperty("token").GetValue(result.Value);
+            Assert.NotNull(token);
+            Assert.True(Regex.IsMatch(token, @"[\S]+\.[\S]+\.[\S]"));
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs]: ^ &

**Review by romanstat — APPROVED** [review]

**Pull request merged**